CMakeLists.txt

```diff
@@ -407,6 +408,15 @@ else()
     )
 endif()
 
+if((UMF_BUILD_GPU_TESTS OR UMF_BUILD_GPU_EXAMPLES) AND UMF_BUILD_CUDA_PROVIDER)
+    find_package(CUDA REQUIRED cuda)
+    set(UMF_CUDA_ENABLED TRUE)
+    # TODO do the same for ze_loader
+else()
+    message(STATUS "Disabling tests and examples that use the CUDA provider "
+                   "because the CUDA libraries they require were not found.")
```

Comment:
It is not true. You got here when `((UMF_BUILD_GPU_TESTS OR UMF_BUILD_GPU_EXAMPLES) AND UMF_BUILD_CUDA_PROVIDER)` is false, but `the CUDA libraries they require were not found` is true when:
```
find_package(CUDA REQUIRED cuda)
if(NOT CUDA_FOUND)
    message(STATUS "the CUDA libraries not found")
endif()
```